pycbc/inference/sampler_emcee.py

```diff
@@ -0,0 +1,995 @@
+        for tk in range(fp.ntemps):
+            try:
+                fp[group.format(tk=tk)][:] = acf[tk,:]
+            except KeyError:
+                # dataset doesn't exist yet, create it
+                fp[group.format(tk=tk)] = acf[tk,:]
+
+    @staticmethod
+    def read_acceptance_fraction(fp, temps=None, walkers=None):
+        """Reads the acceptance fraction from the given file.
+
+        Parameters
+        -----------
+        fp : InferenceFile
+            An open file handler to read the samples from.
+        temps : {None, (list of) int}
+            The temperature index (or a list of indices) to retrieve. If None,
+            acfs from all temperatures and all walkers will be retrieved.
+        walkers : {None, (list of) int}
+            The walker index (or a list of indices) to retrieve. If None,
+            samples from all walkers will be obtained.
+
+        Returns
+        -------
+        array
+            Array of acceptance fractions with shape (requested temps,
+            requested walkers).
+        """
+        group = 'acceptance_fraction/temp{tk}'
+        if temps is None:
+            temps = numpy.arange(fp.ntemps)
+        if walkers is None:
+            wmask = numpy.ones(fp.nwalkers, dtype=bool)
+        else:
+            wmask = numpy.zeros(fp.nwalkers, dtype=bool)
+            wmask[walkers] = True
+        arrays = []
+        for tk in temps:
+            arrays.append(fp[group.format(tk=tk)][wmask])
+        return numpy.vstack(arrays)
+
+    def write_chain(self, fp, max_iterations=None):
+        """Writes the samples from the current chain to the given file. Results
+        are written to: `fp[fp.samples_group/{vararg}/temp{k}/walker{i}]`,
+        where `{vararg}` is the name of a variable arg, `{k}` is a temperature
+        index (smaller = colder), and `{i}` is the index of a walker.
+
+        Parameters
+        -----------
+        fp : InferenceFile
+            A file handler to an open inference file.
+        max_iterations : {None, int}
+            If samples have not previously been written to the file, a new
+            dataset will be created. By default, the size of this dataset will
+            be whatever the length of the sampler's chain is at this point. If
+            you intend to run more iterations, set this value to that size so
+            that the array in the file will be large enough to accomodate
+            future data.
+        """
+        # chain is ntemps x nwalkers x niterations x ndim
+        samples = self.chain
+        ntemps, nwalkers, niterations, _ = samples.shape
+
+        group = fp.samples_group + '/{name}/temp{tk}/walker{wi}'
+
+        # create an empty array if desired, in case this is the first time
+        # writing
+        if max_iterations is not None:
+            if max_iterations < niterations:
+                raise IndexError("The provided max size is less than the "
+                    "number of iterations")
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->